aspnetcore/host-and-deploy/scaling-aspnet-apps/scaling-aspnet-apps.md

```diff
@@ -0,0 +1,352 @@
+1. [Scale and troubleshoot the app](#3-scale-and-troubleshoot-the-app)
+1. [Create the Azure Services](#4-create-the-azure-services)
+1. [Connect the Azure services](#5-connect-the-azure-services)
+1. [Configure and redeploy the app](#6-configure-and-redeploy-the-app)
+
+In some cases, simple ASP.NET Core apps are able to scale without special considerations. However, some apps require additional configurations to utilize certain framework features or architectural patterns, including the following:
+
+* **Secure form submissions**: Razor Pages, MVC and Web API apps often rely on form submissions. By default these apps use cross site forgery tokens and framework data protection services to secure requests. When deployed to the cloud, these apps must be configured to use a managed data protection service in a secure, centralized location.
+
+* **SignalR circuits**: Blazor Server applications require the use of a centralized Azure SignalR service in order to scale properly and securely. These services also utilize the data protection services mentioned previously.
+
+* **Centralized caching or state management services**: Scalable applications may use Azure Cache for Redis to provide distributed caching. Azure storage or database services may be needed to manage state for frameworks such as Orleans, which can assist in writing stateful apps that scale across many different app instances.
+
+> [!NOTE]
+> The steps ahead demonstrate how to properly address these concerns by deploying a scalable app to Azure Container Apps. Most of the concepts in this tutorial also apply when scaling Azure App Service instances.
+
+## 1) Setup the sample project
+
+You can use the GitHub Explorer application to follow along with this tutorial. Clone the application from GitHub using the following command:
+
+```dotnetcli
+git clone "https://github.com/MicrosoftDocs/mslearn-dotnet-debug-visual-studio-app-service.git"
+```
+
+The sample app uses a search form to browse GitHub repositories by name. The form relies on the built-in ASP.NET Core data protection services to handle anti-forgery concerns. By default, when the app scales horizontally on Container Apps, the data protection service will throw an exception. You'll explore and solve this issue in the steps ahead.
+
+#### Test the app
+
+1. Open the project in Visual Studio by double clicking on the `GitHubBrowser.sln` solution file in the project folder.
+1. Launch the app by clicking the run button at the top of Visual Studio. The project includes a Docker file, which means you can click the arrow next to the run button to start the app using either a Docker Desktop setup or the standard ASP.NET Core local web server.
+
+When the app launches in the browser, you can use the search form to browse for GitHub repositories by name.
+
+:::image type="content" source="./media/scaling-app-screenshot.png" alt-text="A screenshot showing the GitHub Explorer app.":::
+
+## 2) Deploy the app to Azure Container Apps
```

Comment:
Don't put numbers in headings. If you're really concerned folks won't follow this in order, tell them to follow in order.